### ISSUE TITLE
Cryptic MySQL Errors, Potentially Mishandling Column Types

### ISSUE THREAD

**Issue by JacobSyndeo** [vapor/mysql#135]
Using:
Core 3.0.0-rc.1.0.1
DatabaseKit 1.0.0-rc.1
Engine 3.0.0-rc.1.0.1
Fluent 3.0.0-rc.1.0.1
FluentMySQL 3.0.0-rc.1.1
MySQL 3.0.0-rc.1.1 (querying a local MySQL 5.7.21 instance on running on macOS High Sierra 10.13.3)
Vapor 3.0.0-rc.1.1

I'm trying to work with a legacy MySQL database table for my new Vapor 3 project. It worked perfectly with Vapor 2, but on V3, it's been nothing but trouble. I've got 79 columns I need to eventually work with, but only 11 I need to use right now (which are defined as properties in my model):

```swift
var id: Int?
var firstName: String = ""
var lastName: String = ""
var nickname: String?
var email: String = ""
var password: String = ""
var bio: String? = ""
var city: String = ""
var state: String = ""
var province: String = ""
var country: String = ""
```

The problem is that Fluent appears to be doing a `SELECT *` query, and if even one of those 79 columns returns something Fluent doesn't like (regardless of whether or not I've defined those columns in my model yet), I'm presented with one of three situations:

• The browser displays: `Oops, something went wrong` with the console saying simply `incorrectValue (EngineServer.swift:126)` (which is just Engine's error display: `self.error(string)`)
• The browser displays: `The packet could not be parsed into valid a response` (I've opened a PR for a grammar fix in #134)
• The request hangs indefinitely (the browser sits loading until I either load a working route or kill the process)

I monitored MySQL's incoming-query log to see exactly what Fluent is asking it to do, but wasn't able to figure much out. I **did** see that when it hangs, it's when it's sending a `PREPARE` to MySQL but failing to send an `EXECUTE`, which of course causes it to not get any data back… and hang.
The database is receiving and running valid queries in the other two cases, though.

I'm using the following code to call the DB, nothing fancy:
```swift
return User.find(userID, on: req).flatMap(to: View.self) { user in
	guard let user = user else {
		throw Abort(.notFound)
	}
	
	...
}
```

It's not even getting to the `guard` section, though.
`userID` is coming in as an int from the URL; I've checked all of that already, stepping through all of it with breakpoints.

Interestingly, certain rows in my table have worked without issues, depending on their contents. I tried to go through them methodically and figure out which columns are the problem, and it **seems** to be related to `VARCHAR`s that allow `null` (that is, they aren't defined with `NOT NULL`) that also contain anything **but** a `null` value, be it an empty string, or normal content.

I can't be positive though. I only found **that** out by creating a row consisting entirely of `NULL`s (in columns that allow `NULL`) and empty strings (in ones that don't allow `NULL`), which worked. I added content to it column by column, refreshing my Vapor app each time, and it **seemed** to consistently break down when a `NULL`able column contained a non-`NULL` value.

Again, these weren't columns I defined in my model, or else I would've tried setting them as optionals. But they're not defined in my model at all, so it's rather frustrating that Fluent is getting tripped up by them. Vapor 2 didn't have this problem, either.

I'm at a loss, and don't really know where to go from here, so I brought this up in the Vapor Slack's help channel, where they suggested I create this issue. I'm not positive that my code isn't at fault here (I'm definitely new to Vapor), but I would imagine that database rows not included in the model being capable of breaking the app would **not** be expected behavior.

Thanks so much for reading, and thanks for the great work on this project so far. Vapor is an amazing framework with a bright `Future` (😉), and I want to help it along however I can. I can provide whatever additional info to help resolve this issue—just let me know!

**Comment by JacobSyndeo** [comment]
Upon reading through some of the closed issues, I came across #116, which seems to have a lot of similarities with this one. I'd be surprised if they weren't related.

**Comment by JacobSyndeo** [comment]
Upon reading through some of the closed issues, I came across #116, which seems to have a lot of similarities with this one (not exactly the same—mine seems to have problems when certain NULLable fields contain something besides NULL, but it still seems similar). I'd be surprised if they weren't related.

**Comment by JacobSyndeo** [comment]
This also might be related to the endless hanging in vapor/fluent-mysql#60, which mentions NIO as a potential fix. But NIO sounds like a pretty sizable overhaul, and I'm not sure if it's even semi-functional yet, so I don't know if testing on that branch would even be possible right now.
